src/main/java/io/quarkus/security/identity/SecurityIdentity.java

```diff
@@ -121,7 +124,7 @@ default boolean hasRole(String role) {
      */
     default boolean checkPermissionBlocking(Permission permission) {
         try {
-            return checkPermission(permission).toCompletableFuture().join();
+            return checkPermission(permission).await().indefinitely();
         } catch (CompletionException e) {
```

Comment:
You should catch `Exception`. 
If it's a RuntimeEXception, the exception is propagated directly.

Reply:
Actually, the try/catch is not required. It's what is happening under the hood.